st2common/st2common/util/concurrency.py

```diff
@@ -111,3 +113,21 @@ def kill(green_thread, *args, **kwargs):
         return green_thread.kill(*args, **kwargs)
     else:
         raise ValueError('Unsupported concurrency library')
+
+
+def sleep(*args, **kwargs):
+    if CONCURRENCY_LIBRARY == 'eventlet':
```

Comment:
Seems like there should be a better way to handle the eventlet vs. gevent driver than to use all these if/else across different methods in this module.

Reply:
I don't think we get much with doing it differently.

We could abstract it some more, in some driver / plugin approach but that would be massive amount of work for very little value.